src/main/scala/com/microsoft/hyperspace/util/ResolverUtils.scala

```diff
@@ -71,4 +76,68 @@ object ResolverUtils {
       availableStrings: Seq[String]): Option[Seq[String]] = {
     Some(requiredStrings.map(resolve(spark, _, availableStrings).getOrElse { return None }))
   }
+
+  /**
+   * Finds all resolved strings for requiredStrings, from the given logical plan. Returns
+   * optional seq of resolved strings if all required strings are resolved, otherwise None.
+   *
+   * @param spark Spark session.
+   * @param requiredStrings List of strings to resolve.
+   * @param plan Logical plan to resolve against.
+   * @return Optional Seq of resolved strings if all required strings are resolved. Else, None.
+   */
+  def resolve(
+      spark: SparkSession,
+      requiredStrings: Seq[String],
+      plan: LogicalPlan): Option[Seq[String]] = {
+
+    def fixParts(parts: Seq[String], outputs: Seq[Attribute]): Seq[String] = {
+      var newParts = Seq.empty[String]
+      val h :: t = parts.toList
+      val topLevelField = outputs.find(_.name.compareToIgnoreCase(h) == 0)
+      topLevelField match {
+        case Some(o) =>
+          newParts = newParts :+ o.name
+          var children = o.dataType.asInstanceOf[StructType]
+          t.foreach { e =>
+            val elIdx: Int = children.fieldNames.indexWhere { f => f.compareToIgnoreCase(e) == 0 }
+            if (elIdx >= 0) {
+              newParts = newParts :+ children.fieldNames(elIdx)
+              children.fields.toSeq(elIdx).dataType match {
+                case s: StructType =>
+                  children = s
+                case _ =>
+              }
+            }
+          }
+        case None =>
+          newParts = h +: t
+      }
+      newParts
+    }
+
+    val schemaFieldNames = plan.output.map(_.name)
+    Some(requiredStrings.map { requiredField =>
+      plan
+        .resolveQuoted(requiredField, spark.sessionState.conf.resolver)
+        .getOrElse { return None }
+        .collectFirst {
+          case a: AttributeReference =>
+            schemaFieldNames.find(_.compareToIgnoreCase(a.name) == 0).getOrElse(a.name)
+          case g: GetStructField =>
+            val parts = g.sql // returns backtick enclosed string (ie: `a`.`b`.`c`)
+              .replaceAll("^`(.*)`$", "$1") // strip first and last backticks
+              .split("`\\.`") // split by "`.`"
+            val fixedParts = fixParts(parts, plan.output) // fix the casing
+            val resolvedFieldName = fixedParts.mkString(".") // put back the field name fixed
+            val unsupported = parts.filter(_.contains("."))
+            if (unsupported.nonEmpty) { // the fields with dots should throw exception
+              throw HyperspaceException(s"The following field name construct " +
+                s"$resolvedFieldName contains unsupported parts: ${unsupported.mkString(",")}.")
```

Comment:
How about this:
`Hyperspace does not support nested column names including '.' (columnName: ${requiredField})` ?

Reply:
This not 100% true. We do support indexes over field names - this what are trying to achieve. We just don't support **not nested** fields having field names with dots.

Reply:
I'll put this message:
```
s"Hyperspace does not support the following access " +
s"field path construct $requiredField because it contains the following " +
s"unsupported field names that contain dots: ${unsupported.mkString(",")}."
```

Let me give an example...

Given a dataset with the following schema:
```
root
 |-- id: integer (nullable = true)
 |-- nested: struct (nullable = true)
 |    |-- the.non.nested.part: struct (nullable = true)
 |    |    |-- cnt: integer (nullable = true)
```
Creating an index with:

```
IndexConfig("idx", indexedColumns = Seq("nested.the.non.nested.part.cnt"))
```

Is not supported and the exception message will look like this:

```
Hyperspace does not support the following access field path nested.the.non.nested.part.cnt because it contains the following unsupported field names that contain dots: the.non.nested.part.
```

Reply:
Thanks for revising it. It looks better 👍

Reply:
Thanks for revising this! it looks better 👍